File: practica-1-grupo-1-equipo-6/src/gestorAplicacion/utileria/Mesa.java
//Clase Mesa
//Autores: Michael Moreno Valoyes
//importante en la funcionalidad de administrar y verificar el estado del restaurante.


package gestorAplicacion.utileria;
import gestorAplicacion.persona.*;
import java.io.Serializable;

//Clase Mesa, desde este se define y cambia el estado de disponibilidad de las mesas presentes en el restaurante.
public class Mesa implements Serializable{
	
	private static final long serialVersionUID = 7L;
	private boolean disponibilidad; //este atributo no solo es cambiado segun reserva sino tambien cuando un cliente que no ha reservado ocupa una mesa.
	private int numero; // Es un atributo identificador
	private int sillas;
	private Cliente cliente;
	
	//constructor
	//considera si la mesa fue reservada o no y la agrega a la lista correspondiente
	public Mesa(boolean disponibilidad, int numero, int sillas) {
		this.disponibilidad = disponibilidad;
		this.numero = numero;
		this.sillas = sillas;
		
		if(disponibilidad) {
			Restaurante.getMesasDisponibles().add(this);
		} else {
			Restaurante.getMesasReservadas().add(this);
		}
	}

	//Metodos get y set para el atributo disponibilidad
	public boolean isDisponibilidad() {
		return disponibilidad;
	}

	public void setDisponibilidad(boolean disponibilidad) {
		this.disponibilidad = disponibilidad;
	}
	
	//Metodos get y set para el atributo numero (que tambien es un identificador)
	public int getNumero() {
		return numero;
	}

	public void setNumero(int numero) {
		this.numero = numero;
	}
	
	//Metodos get y set para el atributo sillas
	public int getSillas() {
		return sillas;
	}

	public void setSillas(int sillas) {
		this.sillas = sillas;
	}
	
	//Metodos get y set para el atributo cliente
	public Cliente getCliente() {
		return cliente;
	}
	public void setCliente(Cliente cliente) {
		this.cliente = cliente;
	}
}

File: practica2/practica2/UI/gestorAplicacion/utileria/mesa.py
# Clase Mesa
# Importante en la funcionalidad de administrar y verificar el estado del restaurante
class Mesa:
    # La crea y considera si la mesa está disponible o no
    def __init__(self, disponibilidad, numero, sillas):
        from gestorAplicacion.utileria.restaurante import Restaurante
        self.disponibilidad = disponibilidad
        self.numero = numero
        self.sillas = sillas
        self.cliente = None
        if self.disponibilidad:
            Restaurante.getMesasDisponibles().append(self)
        else:
            Restaurante.getMesasReservadas.append(self)
    # Sets y gets

    def isDisponibilidad(self):
        return self.disponibilidad

    def setDisponibilidad(self, disponibilidad):
        self.disponibilidad = disponibilidad

    def getNumero(self):
        return self.numero

    def setNumero(self, numero):
        self.numero = numero

    def getSillas(self):
        return self.sillas

    def setSillas(self, sillas):
        self.sillas = sillas

    def getCliente(self):
        return self.cliente

    def setCliente(self, cliente):
        self.cliente = cliente
